File: iOS/TaskBridge/Repository/RemindersRepository.swift
import Foundation
import TaskBridgeCore

@MainActor
protocol RemindersRepository: AnyObject {
    var remindersState: AsyncStream<RemindersState> { get }
    
    func loadReminders() async throws
    func scheduleReminder(_ reminder: Reminder) async throws
    func cancelReminder(_ reminderId: ReminderId) async throws
}

File: Android/src/androidMain/kotlin/com/taskbridge/android/repository/RemindersRepository.kt
package com.taskbridge.android.repository

import com.taskbridge.core.models.reminders.Reminder
import com.taskbridge.core.models.reminders.ReminderId
import com.taskbridge.core.usecases.reminders.RemindersState
import kotlinx.coroutines.flow.StateFlow

/**
 * Repository for managing reminders on Android.
 */
interface RemindersRepository {
    val remindersState: StateFlow<RemindersState>
    
    suspend fun loadReminders()
    suspend fun scheduleReminder(reminder: Reminder)
    suspend fun cancelReminder(reminderId: ReminderId)
}
